src/widget/form/chatform.cpp

```diff
@@ -879,6 +889,52 @@ void ChatForm::loadHistory(QDateTime since, bool processUndelivered)
     chatWidget->verticalScrollBar()->setValue(savedSliderPos);
 }
 
+void ChatForm::onScreenshotCreate()
+{
+    QRect region;
+    int result = 0;
+
+    {
+        ScreenshotDialog screenshot(region);
+        result = screenshot.exec();
+    }
+
+    if (result == QDialog::Accepted)
+    {
+        QScreen *screen = QGuiApplication::primaryScreen();
+        if (screen)
+        {
+            // Have to wait for the rubber band to disappear.
+            // Using flush, sendPostedEvents and processEvents does not work.
+            QTimer::singleShot(100, this, [=]()
+            {
+                QTemporaryFile file("qTox-Screeshot-XXXXXXXX.png");
+
+                if (!file.open())
+                {
+                    QMessageBox::warning(this, tr("File not read"), tr("qTox wasn't able to save the screenshot"));
+                    return;
+                }
+                file.setAutoRemove(false);
+                // QPixmap is copy-on-write. We have to copy to crop anyway.
+                (screen->grabWindow(0).copy(region)).save(&file, "PNG");
+                if (file.isSequential())
```

Comment:
Uh... why would your temporary file ever be sequential? We're not sanitizing user input here, I don't see the purpose